graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/DmlFetcher.java

```diff
@@ -109,24 +110,64 @@ protected DmlFetcher(Table table, NameMapping nameMapping) {
     }
   }
 
-  protected List<Map<String, Object>> toListOfMutationResults(
-      ResultSet resultSet, List<Map<String, Object>> originalValues) {
-    List<Row> rows = resultSet.currentPageRows();
+  protected List<Map<String, Object>> toBatchResults(
+      List<Row> rows, List<Map<String, Object>> originalValues) {
+
+    boolean applied = isAppliedBatch(rows);
+
     List<Map<String, Object>> results = new ArrayList<>();
-    if (rows.isEmpty()) {
-      for (Map<String, Object> originalValue : originalValues) {
-        results.add(toAppliedMutationResultWithOriginalValue(originalValue));
-      }
-    } else {
-      for (int i = 0; i <= rows.size(); i++) {
-        Row row = rows.get(i);
-        Map<String, Object> originalValue = originalValues.get(i);
-        results.add(toMutationResultSingleRow(originalValue, row));
-      }
+    for (Map<String, Object> originalValue : originalValues) {
+      results.add(
+          applied
+              ? toAppliedMutationResultWithOriginalValue(originalValue)
+              : toUnappliedBatchResult(rows, originalValue));
     }
     return results;
   }
 
+  protected Map<String, Object> toBatchResult(List<Row> rows, Map<String, Object> originalValue) {
+    return isAppliedBatch(rows)
+        ? toAppliedMutationResultWithOriginalValue(originalValue)
+        : toUnappliedBatchResult(rows, originalValue);
+  }
+
+  private boolean isAppliedBatch(List<Row> batchRows) {
+    if (batchRows.isEmpty()) {
+      return true;
+    }
+    if (batchRows.size() > 1) {
+      return false;
+    }
+    Row row = batchRows.get(0);
+    return row.columns().stream().map(Column::name).anyMatch("[applied]"::equals)
+        && row.getBoolean("[applied]");
+  }
+
+  private Map<String, Object> toUnappliedBatchResult(
+      List<Row> rows, final Map<String, Object> originalValue) {
+    Map<String, Object> primaryKey =
+        table.primaryKeyColumns().stream()
```

Comment:
isn't the `table` containing all the fields?
What if the graphQL selection does not specify all the fields to be returned?
Are we filling it anyway, but the GraphQL returns only those fields that are in the user's selection?

Reply:
We're filling the response with `row`, which may or may not contain all the fields depending on the original request (this PR doesn't change anything in that regard).

The goal of this method is to pick the correct row, because they are not necessarily in the right order (see `AtomicBatchTest. failedConditionalBatchOutOfOrder`).